src/transactions/utils.ts

```diff
@@ -47,3 +54,43 @@ export function groupFeedItemsInSections(
       data: value.data,
     }))
 }
+
+export function* buildBaseTransactionReceipt(
```

Comment:
I wasn't sure where to put this function, any suggestion?

Reply:
i think the scope of abstraction is a little odd, because it seems like it tries to turn the transaction receipt into the arguments that the reducer accepts except the reducer then needs to take the output of this function and do more logic (buildGasFees) for the fees. 

i have a couple of suggestions, maybe there are others that make sense too:
1. have a handleTransactionConfirmed saga that is responsible for dispatching transactionConfirmed given the receipt (and formats the fees in the format that can be directly saved into redux).
2. have a toNetworkFees saga that is just responsible for getting the network fees format from the transaction receipt

Reply:
Hmm... Creating a new saga for this? It sounds a lot for just a mapper.
What would be the difference between a saga and this? Maybe I can move the `buildGasFees` function to here.

What do you think if I put this logic inside the `transactionConfirmed` action?
I think that's the place to put this logic.
But since all our actions are pure, I didn't want to add code there, also I don't know if I'm able to access the redux state from there, since I need the token list to get some info.

I think we can think this code as a "parser" from the blockchain response into our schema.

Reply:
hum, i'm not sure i understand the concern. what you've got here in buildBaseTransactionReceipt is already a saga? (maybe the concern is that i am using "saga" and "generator function" interchangably?) i'm only proposing that we keep logic in one place rather than having it broken up. most of the logic here i see is trying to massage the tx receipt to be what we want to store in redux. so i thought having a generator function that takes a receipt and calls the `transactionConfirmed` action with the expected params seemed reasonable. 

i think redux actions are meant to be simple and you won't have access to the redux state from there. i also thought maybe this logic could live in the reducer, but we shouldn't pass unserialisable data to the reducers (i believe the viem tx receipts have bigInt's). so i think having a generator function / saga is probably the easiest solution